api/src/main/java/jakarta/nosql/Template.java

```diff
@@ -0,0 +1,279 @@
+ *         .eq("Java Concurrency in Practice")
+ *         .and("author")
+ *         .eq("Brian Goetz")
+ *         .and("year")
+ *         .eq(Year.of(2006))
+ *         .singleResult();
+ *
+ * template.delete(Book.class)
+ *         .where("author")
+ *         .eq("Joshua Bloch")
+ *         .and("edition")
+ *         .gt(3)
+ *         .execute();
+ *
+ * }</pre>
+ *
+ * @see QueryMapper
+ */
+public interface Template {
+
+    /**
+     * Inserts an entity into the database. If an entity of this type with the same
+     * unique identifier already exists in the database and the database supports ACID transactions,
+     * then this method raises an error. In databases that follow the BASE model
+     * or use an append model to write data, this exception is not thrown.
+     *
+     * <p>The entity instance returned as a result of this method must include all values that were
+     * written to the database, including all automatically generated values and incremented values
+     * that changed due to the insert. After invoking this method, do not continue to use the instance
+     * that is supplied as a parameter. This method makes no guarantees about the state of the
+     * instance that is supplied as a parameter.</p>
+     *
+     * @param entity the entity to insert. Must not be {@code null}.
+     * @param <T>    the entity type
+     * @return the inserted entity, which may or may not be a different instance depending on whether the insert
+     * caused values to be generated or automatically incremented.
+     * @throws NullPointerException if the entity is null.
+     */
+    <T> T insert(T entity);
+
+    /**
+     * Inserts an entity into the database with an expiration to the entity. If an entity of this type with the same
+     * unique identifier already exists in the database and the database supports ACID transactions,
+     * then this method raises an error. In databases that follow the BASE model
+     * or use an append model to write data, this exception is not thrown.
+     *
+     * <p>The entity instance returned as a result of this method must include all values that were
+     * written to the database, including all automatically generated values and incremented values
+     * that changed due to the insert. After invoking this method, do not continue to use the instance
+     * that is supplied as a parameter. This method makes no guarantees about the state of the
+     * instance that is supplied as a parameter.</p>
+     *
+     * @param entity the entity to insert. Must not be {@code null}.
+     * @param ttl    time to live
+     * @param <T>    the entity type
+     * @return the inserted entity, which may or may not be a different instance depending on whether the insert caused
+     * values to be generated or automatically incremented.
+     * @throws NullPointerException          if the entity is null.
+     * @throws UnsupportedOperationException when the database does not provide TTL
```

Comment:
What should the user expect to happen if the time-to-live is exceeded? This method doesn't specify.

Reply:
Excellent, now the user will know what to expect of it. I'll mark this conversation resolved.